lib/js/src/rpc/RpcStruct.js

```diff
@@ -88,21 +88,15 @@ class RpcStruct {
     * @return {null|Object}
     */
     formatObject(tClass, obj) {
-        if (obj === null) {
+        if (obj === null || obj === undefined) {
             return null;
         } else if (obj.constructor === tClass) {
             // if tClass is String and obj is a String, this should execute
             return obj;
-        } else if (obj.constructor === String) {
-            if (tClass instanceof Enum) {
-                return tClass.valueForString(obj);
-            } else if (tClass instanceof String) {
-                // this may be redundant
-                return obj;
-            }
-            return null;
+        } else if (obj.constructor === String) { // this covers the Enum case too
+            return obj;
         } else if (obj.constructor === Object) {
-            if (tClass instanceof RpcStruct) {
+            if (tClass.prototype instanceof RpcStruct) {
                 return new tClass(obj);
             }
             return null;
```

Comment:
It looks like I can't suggest a change to unchanged lines but since this is in the same function I feel like this should be fixed as well. The item isn't declared and it should be the value in the array and not the index of the array. So the last part would look like.
```          
 } else if (obj.constructor === Array) {
            // TODO: ensure completeness
            if (obj.length > 0) {
                const outArray = [];
                for (const item of obj) {
                    outArray.push(this.formatObject(tClass, item));
                }
                return outArray;
            }
        }
```